keras_applications/resnet_common.py

```diff
@@ -227,12 +228,17 @@ def block3(x, filters, kernel_size=3, stride=1, groups=32,
     x = layers.ZeroPadding2D(padding=((1, 1), (1, 1)), name=name + '_2_pad')(x)
     x = layers.DepthwiseConv2D(kernel_size, strides=stride, depth_multiplier=c,
                                use_bias=False, name=name + '_2_conv')(x)
-    x_shape = backend.int_shape(x)[1:-1]
-    x = layers.Reshape(x_shape + (groups, c, c))(x)
-    output_shape = x_shape + (groups, c) if backend.backend() == 'theano' else None
-    x = layers.Lambda(lambda x: sum([x[:, :, :, :, i] for i in range(c)]),
-                      output_shape=output_shape, name=name + '_2_reduce')(x)
-    x = layers.Reshape(x_shape + (filters,))(x)
+    if backend.backend() == 'theano':
+        x_shape = backend.int_shape(x)[1:-1] + (filters,)
+    else:
+        x_shape = None
+    kernel = np.zeros((1, 1, filters * c, filters), dtype=np.float32)
+    for i in range(filters):
+        start = (i // c) * c * c + i % c
+        end = start + c * c
+        kernel[:, :, start:end:c, i] = 1.
+    x = layers.Lambda(lambda x: backend.conv2d(x, backend.variable(kernel)),
```

Comment:
The problem with using a Lambda, is that they are only serializable via bytecode, which will severely limit the portability of the model (for instance, it could not be exported to TF.js or to CoreML).

An alternative:

Use a `Conv2D` layer with `use_bias=False` and `trainable=False`. Set `weights=[kernel]` in the constructor.
Make its `kernel` value part of the weights file being loaded into the model (so that we don't break weight loading).

What do you think?

Reply:
@fchollet, Yes it is true. I think it is better to remove a Lambda, too. The reason why I revised with the Lambda is to avoid updating pretrained weights. In other words, if `Lambda` is replaced by `Conv2D`, `model = ResNeXt50()` raises `ValueError: You are trying to load a weight file containing 107 layers into a model with 123 layers`. E.g.,
```python
# x = layers.Lambda(lambda x: backend.conv2d(x, backend.variable(kernel)),
#                   output_shape=x_shape, name=name + '_2_reduce')(x)
x = layers.Conv2D(filters, 1, use_bias=False, trainable=False, name=name + '_2_reduce')(x)
```

Do you think the Conv2D version with updating pretrained weights is a better choice? I will follow your opinion.